types/next-server/router.d.ts

```diff
@@ -96,5 +96,7 @@ export function withRouter<T extends {}, Q = DefaultQuery>(
   Component: React.ComponentType<T & WithRouterProps<Q>>
 ): React.ComponentType<Omit<T, keyof WithRouterProps<Q>>>;
 
+export function useRouter(): RouterProps;
```

Comment:
```suggestion
export function useRouter<Q = DefaultQuery>(): RouterProps<Q>;
```

I think we should still include a generic here for passing the router query.